### ISSUE TITLE
Bug: "Action is Unknown" Error for GitHub Enterprise Server (GHES) Self-Hosted Actions

### ISSUE THREAD

**Issue by ahenriksen-inferno** [dsoftwareinc/ghactions-manager#191]
### Describe the bug

The plugin does not currently support GH Actions hosted on a self-hosted GHES instance. When opening a workflow that contains a GHA from a GHES instance, the JetBrains IDE / GitHub Actions Manager plugin reports an error for the GHA: `Action example-org/example-gha-name is unknown`. I believe this is due to GHES self-hosted instances using different hostnames.

Example GHES URLs:

- Web: `https://octocorp.ghe.com/org/repo`
- API: `https://octocorp.ghe.com/api/v3`
- GraphQL: `https://octocorp.ghe.com/api/graphql`

References:

- [GHES Docs](https://docs.github.com/en/enterprise-server)
- [GHES Rest API Docs](https://docs.github.com/en/enterprise-server/rest)
- [GHES GraphQL API Docs](https://docs.github.com/en/enterprise-server/graphql)
- [Similar Issue in another JetBrains GHA plugin](https://github.com/YunaBraska/github-workflow-plugin/issues/52) - This may be useful as one of the users has provided some examples and possible solutions for the developer who is unfortunately unable to maintain the plugin further.

Screenshot of error:

<img width="1001" alt="Screenshot of Error in WebStorm" src="https://github.com/user-attachments/assets/8ccf964b-f4db-44e7-9d6d-062a66738ec2">

### Steps to reproduce

1. Create a GitHub worfklow file.
2. Add a GHA from a self-hosted GHES instance to the workflow file.
3. IDE will report `Action is unknown`.

### Expected behavior

The GitHub Actions Manager plugin should support GHAs hosted on GHES instances. If helpful, the host and token for my GHES instance are configured in `Settings -> Version Control -> GitHub`. Or allow specifying other GHA hosts in the plugin configuration. 

### Additional context

Plugin version: 2024.2.13
IDE: WebStorm 2024.2.4
OS: macOS Sequoia 15.1 (24B83)

## Upvote & Fund

- We're using [Polar.sh](https://polar.sh/dsoftwareinc) so you can upvote and help fund this issue.
- We receive the funding once the issue is completed & confirmed by you.
- Thank you in advance for helping prioritize & fund our backlog.

<a href="https://polar.sh/dsoftwareinc/ghactions-manager/issues/191">
<picture>
  <source media="(prefers-color-scheme: dark)" srcset="https://polar.sh/api/github/dsoftwareinc/ghactions-manager/issues/191/pledge.svg?darkmode=1">
  <img alt="Fund with Polar" src="https://polar.sh/api/github/dsoftwareinc/ghactions-manager/issues/191/pledge.svg">
</picture>
</a>

**Comment by olmoser** [comment]
I'm having the same issue but with a local action file reference:

```
---
name: deploy cloud function DEV

on:
  push:
    branches:
      - dev

jobs:
  create-pr:
    uses: ./.github/workflows/create-pr.yml
    secrets: inherit
    with:
      branch: test
      environment: dev
```

![image](https://github.com/user-attachments/assets/84d7754b-5fb2-4da9-b545-3f7b510cf077)

**Comment by ahenriksen-inferno** [comment]
@cunla - Thank you for replying. My project uses a single GHES repository, but in the GitHub plugin settings I have multiple accounts configured because I have projects and NPM packages from both GitHub.com and my GHES server. For example, my accounts are configured as:

- Account 1: GitHub.com account (set as default)
- Account 2: GitHub.com account
- Account 3: GHES account (project's repo account)

I've found that if I don't set a GitHub.com account as the default, the IDE (GHA plugin?) won't be able to retrieve information about the GitHub.com Actions in my workflow.

**Comment by cunla** [comment]
@olmoser can you open a separate issue for your case? it is not related.

**Comment by cunla** [comment]
@ahenriksen-inferno, thanks for the quick response.

This info helps. 
There might be an issue with the way it picks the server-path for repo.
I will check.

**Comment by olmoser** [comment]
#192 @cunla

**Comment by cunla** [comment]
Resolved in 2024.2.14 - will be released soon.

**Comment by ahenriksen-inferno** [comment]
@cunla Awesome! Thank you so much!
